core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/TupleCodec.java

```diff
@@ -161,70 +161,90 @@ public TupleValue parse(@Nullable String value) {
     }
 
     TupleValue tuple = cqlType.newValue();
+    int length = value.length();
 
     int position = ParseUtils.skipSpaces(value, 0);
     if (value.charAt(position) != '(') {
       throw new IllegalArgumentException(
           String.format(
-              "Cannot parse tuple value from \"%s\" at character %d: expecting '(' but got '%c'",
+              "Cannot parse tuple value from \"%s\", at character %d expecting '(' but got '%c'",
               value, position, value.charAt(position)));
     }
 
     position++;
     position = ParseUtils.skipSpaces(value, position);
 
-    if (value.charAt(position) == ')') {
-      return tuple;
-    }
-
     CodecRegistry registry = cqlType.getAttachmentPoint().getCodecRegistry();
 
-    int i = 0;
-    while (position < value.length()) {
+    int field = 0;
+    while (position < length) {
+      if (value.charAt(position) == ')') {
+        position++;
+        if (position == length) {
+          return tuple;
+        } else {
+          position = ParseUtils.skipSpaces(value, position);
+          if (position == length) {
+            return tuple;
+          }
+          throw new IllegalArgumentException(
+              String.format(
+                  "Cannot parse tuple value from \"%s\", at character %d expecting EOF or blank, but got \"%s\"",
+                  value, position, value.substring(position)));
+        }
+      }
```

Comment:
This can be simplified a little bit, given that `skipSpaces` is a no-op if there are no spaces:
```java
    while (position < length) {
      if (value.charAt(position) == ')') {
        position = ParseUtils.skipSpaces(value, position + 1);
        if (position == length) {
          return tuple;
        }
        throw new IllegalArgumentException(
            String.format(
                "Cannot parse tuple value from \"%s\", at character %d expecting EOF or blank, but got \"%s\"",
                value, position, value.substring(position)));
      }
      int n;
```